xcode/Subconscious/Shared/UIKit/BlockEditor/BlockEditorViewController.swift

```diff
@@ -0,0 +1,1211 @@
+//
+//  BlockEditorViewController.swift
+//  BlockEditor
+//
+//  Created by Gordon Brander on 7/17/23.
+//
+
+import UIKit
+import Combine
+import os
+
+extension BlockEditor {
+    /// Represents the sections in the collection view.
+    enum Section: Int, CaseIterable {
+        case blocks = 0
+        case appendix = 1
+    }
+}
+
+extension BlockEditor {
+    /// Manages the "DOM" of the editor.
+    class ViewController:
+        UIViewController,
+        UICollectionViewDelegateFlowLayout,
+        UICollectionViewDataSource
+    {
+        typealias Store = ControllerStore.Store<BlockEditor.ViewController>
+        
+        static var logger = Logger(
+            subsystem: Config.default.rdns,
+            category: "BlockEditorViewController"
+        )
+        
+        /// Create a configured collection view suitable for our block editor
+        private static func createCollectionView(
+            frame: CGRect
+        ) -> UICollectionView {
+            var config = UICollectionLayoutListConfiguration(
+                appearance: .plain
+            )
+            config.showsSeparators = false
+            let layout = UICollectionViewCompositionalLayout.list(
+                using: config
+            )
+            
+            let collectionView = UICollectionView(
+                frame: frame,
+                collectionViewLayout: layout
+            )
+            collectionView.contentInset = UIEdgeInsets(
+                top: 8,
```

Comment:
`AppTheme.unit2`?

Reply:
Done

Reply:
And thanks for catching. I started prototyping this outside of Subconscious, so I didn't have the AppTheme constants at hand. Forgot to add them when I ported over!